### ISSUE TITLE
Sleep behaves differently on Connected/Modern Standby

### ISSUE THREAD

**Issue by Gaurav-Original-ClassicShellTester** [Open-Shell/Open-Shell-Menu#719]
On PCs which have Modern Standby/Connected Standby, Sleep command from Open Shell menu is making them turn off (or hibernate? I don't know) but it's definitely not entering just Sleep. Running Windows 10 20H2.

**Comment by axobarcala** [comment]
TL;DR: The behavior of Open-Shell’s sleep button varies depending upon the enabled sleep state(s). On systems with enabled "Modern Standby" (S0 Low Power Idle) the sleep state can be invoked by turning off the display programmatically [0] [1]

This issue already has been raised here: https://github.com/Open-Shell/Open-Shell-Menu/issues/387

I’ll try to describe the issue in detail, hopefully this should allow the devs to reproduce the issue this time. Since the sleep state behavior highly differentiates between system configurations (OS/BIOS, settings/versions), this might only apply to systems similar to my setup (Dell XPS 13 9365 on Win10 20H2).

Check if  "Modern Standby" is enabled on system:

```
C:\Windows\System32>powercfg /a
The following sleep states are available on this system:
    Standby (S0 Low Power Idle) Network Disconnected
    Hibernate
<truncated>
```

Behavior with "Modern Standby" and "Hibernate" available (=enabled) on the system:

- If "Sleep" is invoked from the Open-Shell menu the system will enter "Hibernate"
- If "Sleep" is invoked from the Windows Start Menu the system will enter "Modern Standby"

Behavior with only "Modern Standby" available (=enabled) on the system:

- If "Sleep" is invoked from the Open-Shell menu the system will do nothing
- If "Sleep" is invoked from the Windows Start Menu the system will enter "Modern Standby"

Note: Just to clarify, this mess is due to the poor "Modern Standby" implementation by MS. Open-Shell’s behavior to enter standby, hibernate or do nothing, depending upon the sleep state configuration, is consistent with other tools, e.g. "nircmd standby" or just running "rundll32.exe powrprof.dll,SetSuspendState 0,1,0" from the command line.

Prepare workaround:

- Check if the system is able to enter "Modern Standby" from command shell. Run the following in the command shell (cmd.exe not powershell)
`powershell (Add-Type '[DllImport(\"user32.dll\")]^public static extern int SendMessage(int hWnd, int hMsg, int wParam, int lParam);' -Name a -Pas)::SendMessage(-1,0x0112,0xF170,2)`

Note: This is the same as using the "Sleep" command from the Windows Start Menu. The system should turn off the display and a few seconds later enter standby. It’s necessary that no devices (e.g. mouse) are prohibiting the system from entering standby. Entering "Modern Standby", at least on my system, can be super finicky, due to a attached mouse. Apart from the obvious, to check if the system entered the sleep state successfully, you might want to run "powercfg -SleepStudy" afterwards.

Workaround:

- Create a batch file with the command from above. Add a menu item in Open-Shell linking to the batch file.

Proposed solution:

- Add an option to the Open-Shell shutdown menu to replace the currently used standby function with a display sleep function for the "Sleep" button.


Ref:
[0] [putting-pc-into-sleep-mode-programmatically](https://stackoverflow.com/questions/41200314/putting-pc-into-sleep-mode-programmatically)
[1] [turn-off-display-in-windows-on-command](https://superuser.com/questions/321342/turn-off-display-in-windows-on-command)

**Comment by ge0rdi** [comment]
Should be fixed in [latest](https://github.com/Open-Shell/Open-Shell-Menu/releases) build.
